Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -1900,6 +1906,227 @@ RefPtr<WebCore::Icon> WebExtension::bestIconVariantForManifestKey(const JSON::Ob
+            if (commandIdentifier.isEmpty()) {
+                error = WEB_UI_STRING("Empty or invalid identifier in the `commands` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for invalid command identifier");
+                continue;
+            }
+
+            RefPtr commandObject = commandsObject->getObject(commandIdentifier);
+            if (!commandObject || !commandObject->size()) {
+                error = WEB_UI_STRING("Empty or invalid command in the `commands` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for invalid command");
+                continue;
+            }
+
+            CommandData commandData;
+            commandData.identifier = commandIdentifier;
+            commandData.activationKey = emptyString();
+            commandData.modifierFlags = { };
+
+            bool isActionCommand = false;
+            if (supportsManifestVersion(3) && commandData.identifier == "_execute_action"_s)
+                isActionCommand = true;
+            else if (!supportsManifestVersion(3) && (commandData.identifier == "_execute_browser_action"_s || commandData.identifier == "_execute_page_action"_s))
+                isActionCommand = true;
+
+            if (isActionCommand && !hasActionCommand)
+                hasActionCommand = true;
+
+            // Descriptions are required for standard commands, but are optional for action commands.
+            auto description = commandObject->getString(commandsDescriptionKeyManifestKey);
+            if (description.isEmpty() && !isActionCommand) {
+                error = WEB_UI_STRING("Empty or invalid `description` in the `commands` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for invalid command description");
+                continue;
+            }
+
+            if (isActionCommand && description.isEmpty()) {
+                description = displayActionLabel();
+                if (description.isEmpty())
+                    description = displayShortName();
+            }
+
+            commandData.description = description;
+
+            if (RefPtr suggestedKeyObject = commandObject->getObject(commandsSuggestedKeyManifestKey)) {
+#if PLATFORM(MAC) || PLATFORM(IOS_FAMILY)
+                const auto macPlatform = "mac"_s;
+                const auto iosPlatform = "ios"_s;
+#elif PLATFORM(GTK) || PLATFORM(WPE)
+                const auto linuxPlatform = "linux"_s;
+#endif
+                const auto defaultPlatform = "default"_s;
+
+                String platformShortcut;
+#if PLATFORM(MAC)
+                platformShortcut = !suggestedKeyObject->getString(macPlatform).isEmpty() ? suggestedKeyObject->getString(macPlatform) : suggestedKeyObject->getString(iosPlatform);
+#elif PLATFORM(IOS_FAMILY)
+                platformShortcut = !suggestedKeyObject->getString(iosPlatform).isEmpty() ? suggestedKeyObject->getString(iosPlatform) : suggestedKeyObject->getString(macPlatform);
```

Comment:
I think we want to simply null check the string. That way a platform can be defined as empty with no fallback.

```suggestion
                platformShortcut = suggestedKeyObject->getString(macPlatform) ?: suggestedKeyObject->getString(iosPlatform);
#elif PLATFORM(IOS_FAMILY)
                platformShortcut = suggestedKeyObject->getString(iosPlatform) ?: suggestedKeyObject->getString(macPlatform);
```

Reply:
as far as I can tell WTF::String doesn't have a default boolean operator, so we can't simply null-check the strings, we have to check isNull or isEmpty

Reply:
Oh right, only explicit bool operator works with String.